edb/schema/delta.py

```diff
@@ -2176,12 +2176,45 @@ def _get_expr_field_ast(
         parent_cls = parent_op.get_schema_metaclass()
         has_shadow = parent_cls.get_field(f'orig_{field.name}') is not None
 
-        if context.descriptive_mode:
-            # When generating AST for DESCRIBE AS TEXT, we want
-            # to use the original user-specified and unmangled
-            # expression to render the object definition.
-            expr_ql = edgeql.parse_fragment(self.new_value.origtext)
-        else:
+        expr_ql = None
+        if context.descriptive_mode or context.declarative:
+            # Potentially we want to use the original expression
+            # instead of the normalized one, but this can only be done
+            # safely if the normalized form has sets the same module
+            # as the one that hosts the object we're in.
+            #
+            # We also make a special exception for indexes which
```

Comment:
The comment here seems to be outdated.  Also, indexes *can* use functions, the scope of index expressions is exactly like that of constraints.